### PR TITLE
Apcon modules

### PR DESCRIPTION
**Pull request by davidlee-ap**
##### SUMMARY

Add new modules to support APCON network devices

##### ISSUE TYPE

- New Module Pull Request

##### COMPONENT NAME

lib/modules/network/apconos
lib/module_utils/network/apconos
lib/plugins/action/apconos_config.py
lib/plugins/action/apconos.py
lib/plugins/cliconf/apconos.py
lib/plugins/terminal/apconos.py

##### ADDITIONAL INFORMATION

```paste below

```

### CODE REVIEW

File: test/sanity/ansible-doc/skip.txt
Comment (mattclay) [inline test/sanity/ansible-doc/skip.txt]: New modules should not be added to the skip list.

**Review by mattclay — DISMISSED** [review]

File: test/sanity/ansible-doc/skip.txt
Comment (davidlee-ap) [reply test/sanity/ansible-doc/skip.txt]: I removed it.
Thanks!

Comment (davidlee-ap) [reply test/sanity/ansible-doc/skip.txt]: > 
> 
> New modules should not be added to the skip list.

I removed it.
Thanks!

File: lib/ansible/modules/network/apconos/apconos_cert.py
Comment (kbreit) [inline lib/ansible/modules/network/apconos/apconos_cert.py]: It could also be 2.9 as no new modules are added on minor releases.

File: lib/ansible/modules/network/apconos/apconos_cert.py
Comment (kbreit) [inline lib/ansible/modules/network/apconos/apconos_cert.py]: Capitalize properly

File: lib/ansible/modules/network/apconos/apconos_cert.py
Comment (kbreit) [inline lib/ansible/modules/network/apconos/apconos_cert.py]: Capitalize properly

File: lib/ansible/modules/network/apconos/apconos_cert.py
Comment (kbreit) [inline lib/ansible/modules/network/apconos/apconos_cert.py]: Why is it here then?

File: lib/ansible/modules/network/apconos/apconos_cert.py
Comment (kbreit) [inline lib/ansible/modules/network/apconos/apconos_cert.py]: Why would this be a list instead of a `str`? Also, other modules normally use camel case. I'll defer to a core developer to see what they think about this convention.

File: lib/ansible/modules/network/apconos/apconos_cert.py
Comment (kbreit) [inline lib/ansible/modules/network/apconos/apconos_cert.py]: Do you have anywhere in your code that's supporting check mode?

File: lib/ansible/plugins/action/apconos_config.py
Comment (kbreit) [inline lib/ansible/plugins/action/apconos_config.py]: Copyright is old.

Comment (davidlee-ap) [reply lib/ansible/plugins/action/apconos_config.py]: I'll fix it. Thanks!

File: lib/ansible/modules/network/apconos/apconos_cert.py
Comment (davidlee-ap) [reply lib/ansible/modules/network/apconos/apconos_cert.py]: Sanity test couldn't pass without it. It might be implemented in the future.

File: lib/ansible/modules/network/apconos/apconos_cert.py
Comment (davidlee-ap) [reply lib/ansible/modules/network/apconos/apconos_cert.py]: I'll fix it. Thanks!

File: lib/ansible/modules/network/apconos/apconos_cert.py
Comment (davidlee-ap) [reply lib/ansible/modules/network/apconos/apconos_cert.py]: I'll fix it.
Thanks!

File: lib/ansible/modules/network/apconos/apconos_cert.py
Comment (davidlee-ap) [reply lib/ansible/modules/network/apconos/apconos_cert.py]: I'll fix it!
Thanks!

File: lib/ansible/modules/network/apconos/apconos_cert.py
Comment (davidlee-ap) [reply lib/ansible/modules/network/apconos/apconos_cert.py]: str should be OK, but it is supposed to be a list for the application.

Comment (kbreit) [reply lib/ansible/modules/network/apconos/apconos_cert.py]: What do you mean? How is this interfacing and what’s the application?

File: lib/ansible/modules/network/apconos/apconos_cert.py
Comment (mattclay) [reply lib/ansible/modules/network/apconos/apconos_cert.py]: If it's not used by the module, remove it from the module spec instead of adding it to the docs.

File: lib/ansible/modules/network/apconos/apconos_cert.py
Comment (mattclay) [reply lib/ansible/modules/network/apconos/apconos_cert.py]: The module is only using `ipaddress[0]`, so this shouldn't be a list.

File: lib/ansible/modules/network/apconos/apconos_cert.py
Comment (davidlee-ap) [reply lib/ansible/modules/network/apconos/apconos_cert.py]: I'll remove it.
Thanks!

File: lib/ansible/modules/network/apconos/apconos_cert.py
Comment (davidlee-ap) [reply lib/ansible/modules/network/apconos/apconos_cert.py]: > 
> 
> The module is only using `ipaddress[0]`, so this shouldn't be a list.

I'll fix it.
Thanks!

**Comment by danielmellado** [comment]
It'd be also great to have integration and unit tests. You can check for example /ansible/ansible/tree/devel/test/units/modules/network

**Pull request closed without merging**
